src/PrestaShopBundle/Kernel/ModuleRepositoryFactory.php

```diff
@@ -0,0 +1,155 @@
+ */
+class ModuleRepositoryFactory
+{
+    /**
+     * @var self
+     */
+    private static $instance;
+
+    /**
+     * @var string
+     */
+    private $parameters;
+
+    /**
+     * @var string
+     */
+    private $environment;
+
+    /**
+     * @var ModuleRepository
+     */
+    private $moduleRepository;
+
+    /**
+     * @var string
+     */
+    private $parametersFile = '';
+
+    public static function getInstance()
+    {
+        if (!self::$instance) {
+            self::$instance = new ModuleRepositoryFactory();
+        }
+
+        return self::$instance;
+    }
+
+    /**
+     * @param string|null $environment
+     * @param array|null $parameters
+     */
+    public function __construct($environment = null, array $parameters = null)
+    {
+        $this->environment = $environment;
+        if (null === $environment) {
+            $this->environment = (new Environment())->getName();
+        }
+        $this->parameters = $parameters;
+    }
+
+    /**
+     * @return ModuleRepository
+     *
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    public function getRepository()
+    {
+        if (null !== $this->getParameters() && null === $this->moduleRepository) {
+            $databasePrefix = $this->getParameters()['database_prefix'];
+            $this->moduleRepository = new ModuleRepository(
+                $this->getConnection(),
+                $databasePrefix
+            );
+        }
+
+        return $this->moduleRepository;
+    }
+
+    /**
+     * @return \Doctrine\DBAL\Connection
+     *
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    private function getConnection()
+    {
+        $parameters = $this->getParameters();
+
+        return DriverManager::getConnection(array(
+            'dbname' => $parameters['database_name'],
+            'user' => $parameters['database_user'],
+            'password' => $parameters['database_password'],
+            'host' => $parameters['database_host'],
+            'port' => $parameters['database_port'],
+            'charset' => 'utf8',
+            'driver' => 'pdo_mysql',
+        ));
+    }
+
+    /**
+     * @return string file path to PrestaShop configuration parameters
+     */
+    private function getParametersFile()
+    {
+        if (empty($this->parametersFile) && file_exists(__DIR__ . '/../../../app/config/parameters.php')) {
+            $this->parametersFile = realpath(__DIR__ . '/../../../app/config/parameters.php');
+        }
+
+        return $this->parametersFile;
+    }
+
+    /**
+     * @return array
+     */
+    private function getParameters()
+    {
+        if (null === $this->parameters && !empty($this->getParametersFile())) {
+            $config = require $this->getParametersFile();
+            if ('test' === $this->environment) {
+                $config['parameters']['database_name'] = 'test_' . $config['parameters']['database_name'];
```

Comment:
Not sure forcing the test database with `test_` is a good idea, we should be able to defined our test database.